stablehlo/dialect/VhloOps.cpp

```diff
@@ -333,6 +333,20 @@ LogicalResult verifyConstraint_0_17_0(mlir::Operation* op,
     return failure();
   return success();
 }
+
+LogicalResult verifyConstraint_1_2_1(mlir::Operation* op,
+                                     Version targetVersion) {
+  auto customCallOp = dyn_cast<mlir::vhlo::CustomCallOpV1>(op);
```

Comment:
Probably don't need `mlir::vhlo` I believe we're in that namespace. Also this will only ever be called on CustomCallOp so use `cast`:

```
auto customCallOp = cast<CustomCallOpV1>(op);
if (targetVersion < Version(1, 2, 1) && customCallOp...) { return failure(); }
return success();
```

Also I'm not sure how enum attrs work, do we need the cast or does the following work as an attr comparison?

```
(customCallOp.getApiVersion() == CustomCallApiVersionV1::API_VERSION_TYPED_FFI)
```

Reply:
yeh, mlir::vhlo namespace path is not required here. 

`getApiVersion() returns `mlir::attribute` for vhlo op, not `CustomCallApiVersionV1` enum. Stablehlo op, does return `CustomCallApiVersion` enum for `getApiVersion()`

Reply:
Done

Reply:
> direct attr comparison did not work.

Makes sense, we'd really need to be comparing to:

```
vhlo::CustomCallApiVersionV1Attr::get(getContext(),vhlo::CustomCallApiVersionV1::API_VERSION_TYPED_FFI)
```

But I don't think we should create a new attr just for the sake of comparison, lgtm as-is!